### PR TITLE
AWS S3 key tab

### PR DESCRIPTION
**Pull request by Jin1L**

### CODE REVIEW

**Comment by yassine-moumen** [comment]
## To access Flojoy-studio build from this pull request, you can use the following URLs: 

Windows installation: http://597_windows.flojoy-test.com:3000 
Linux installation: http://597_linux.flojoy-test.com:3000 

Please note that it may take a few minutes for the instance to start up, so please be patient. 

It is important to know that the instance will automatically shut down after 4 hours to conserve resources. If you wish to start the instance again after it shuts down (or restart it to pickup new changes), you can use this API : 

Windows: https://kgpsoscpc4q52d4mtwuu7jhvie0qdiso.lambda-url.us-east-1.on.aws/start_instance?instance_id=i-055523d8e67cef82a&domain=597_windows 

Linux: https://kgpsoscpc4q52d4mtwuu7jhvie0qdiso.lambda-url.us-east-1.on.aws/start_instance?instance_id=i-097b168235adac7e1&domain=597_linux

File: server/views.py [hunk 1]
```diff
@@ -89,3 +89,21 @@ def set_user_api_key(request):
...
+@api_view(["POST"])
+def set_s3_key(request):
+    key = request.data
+    s3_name = key["name"]
+    access_key = key["accessKey"]
+    secret_key = key["secretKey"]
+    set_frontier_s3_key(s3_name, access_key, secret_key)
+
+    response = {
+        "success": True,
```
Comment (JeffDotPng) [inline server/views.py:103]: Don't need to send a success flag in the response, this is already communicated by the 200 status code

File: src/feature/flow_chart_panel/views/S3KeyModal.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    marginTop: 5,
+  },
+}));
+const S3KeyModal = ({ isOpen, onClose }: S3KeyModelProps) => {
+  const { classes } = useStyles();
+  const [s3Name, setS3Name] = useState<string>("");
+  const { s3AccessKey, setS3AccessKey, s3SecretKey, setS3SecretKey } =
+    useFlowChartState();
+
+  const handlNameChange = (e: ChangeEvent<HTMLInputElement>) => {
```
Comment (JeffDotPng) [inline src/feature/flow_chart_panel/views/S3KeyModal.tsx:90]: Typo

File: src/services/FlowChartServices.ts [hunk 3]
```diff
@@ -62,6 +62,49 @@ export const sendApiKeyToDjango = async (apiKey: string) => {
...
+    if (response.ok) {
+      const responseData = await response.json();
+      notifications.update({
+        id: "set-s3-key",
+        title: "Successful!",
+        message: "Successfully set the Access Key",
+        autoClose: 5000,
+      });
+    } else {
+      notifications.update({
```
Comment (JeffDotPng) [inline src/services/FlowChartServices.ts:92]: I think just throwing an error here would work to make it go to the catch block, makes for less repeating code.

**Review by JeffDotPng — APPROVED** [review]

**Pull request merged**